### PR TITLE
Explain saucectl architecture

### PR DESCRIPTION
**Pull request by alexplischke**
### Description

Add an `Architecture` section for saucectl that we can expand slowly over time.

To begin with, this section includes a simplified component diagram, along with a description of the basic worklow.

### CODE REVIEW

**Review by mhan83 — APPROVED** [review]

**Review by tianfeng92 — APPROVED** [review]

File: docs/dev/cli/saucectl/architecture.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+id: architecture
+title: saucectl Architecture
+sidebar_label: Architecture
+---
+
+This document provides an overview of the `saucectl` CLI architecture to help you better understand how it works.
+
+## What You'll Need
```
Comment (lysannep) [inline docs/dev/cli/saucectl/architecture.md:9]: Is this heading needed? The reference to 'Using the saucectl CLI' could be moved to the bottom of the page as a link "for more information".

File: docs/dev/cli/saucectl/architecture.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+If you're new to `saucectl`, we recommend reviewing the [Using the saucectl CLI](/dev/cli/saucectl) documentation.
+
+## Architecture
```
Comment (lysannep) [inline docs/dev/cli/saucectl/architecture.md:13]: The H2 heading for "Architecture" seems a bit redundant/repetitive as it's included in the title/H1 heading.
```suggestion
## Architecture Components
```

File: docs/dev/cli/saucectl/architecture.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+Here's what happens in a nutshell:
+
+1. `saucectl` sends the test payload (mobile app, test app, or project files) to the app storage.
+2. Depending on the capabilities of the underlying test framework (say, Playwright or Espresso), `saucectl` will call the appropriate device cloud that is capable of running your tests.
```
Comment (lysannep) [inline docs/dev/cli/saucectl/architecture.md:18]:
```suggestion
2. Depending on the capabilities of the underlying test framework (e.g., Playwright or Espresso), `saucectl` will call the appropriate device cloud that is capable of running your tests.
```

File: docs/dev/cli/saucectl/architecture.md (hunk 1 above)
Comment (aliciakrebs) [reply docs/dev/cli/saucectl/architecture.md]: Moved to end of page.

File: docs/dev/cli/saucectl/architecture.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+1. `saucectl` sends the test payload (mobile app, test app, or project files) to the app storage.
+2. Depending on the capabilities of the underlying test framework (e.g., Playwright or Espresso), `saucectl` will call the appropriate device cloud that is capable of running your tests.
+3. The device cloud allocates a device for your request, retrieves the payload from app storage, and runs the tests.
+
+This diagram illustrates at high level how `saucectl` interacts with the Sauce Labs environment.
+
+<img src="/img/saucectl/components.jpg" />
+
+More Information
```
Comment (lysannep) [inline docs/dev/cli/saucectl/architecture.md:21]:
```suggestion
## More Information
```

File: docs/dev/cli/saucectl/architecture.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+* [Using the saucectl CLI](/dev/cli/saucectl).
```
Comment (lysannep) [inline docs/dev/cli/saucectl/architecture.md:22]:
```suggestion
* [Using the saucectl CLI](/dev/cli/saucectl)
```

**Review by lysannep — APPROVED** [review]

**Pull request merged**